Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,26 +68,29 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
+          if stanza == :artifacts
+            result = {}
+            value.each do |e|
+              result[e.class.dsl_key] = e.summarize
+            end
+            value = result
```

Comment:
You can replace all of this with

```ruby
value = Hash[value.map { |v| [v.class.dsl_key, v] }]
```

It shouldn't use `#summarize` because otherwise it will not respond to `#to_a` or `#to_h` anymore below.